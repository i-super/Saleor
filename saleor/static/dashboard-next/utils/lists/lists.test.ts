import {
  add,
  addAtIndex,
  isSelected,
  move,
  remove,
  removeAtIndex,
  toggle,
  updateAtIndex
} from "./lists";

const initialArray = ["lorem", "ipsum", "dolor"];

describe("Properly calculates output arrays", () => {
  it("Adds", () => {
    expect(add("sit", initialArray)).toMatchSnapshot();
  });

  it("Adds at index", () => {
    expect(addAtIndex("sit", initialArray, 2)).toMatchSnapshot();
  });

  it("Updates at index", () => {
    expect(updateAtIndex("amet", initialArray, 1)).toMatchSnapshot();
  });

  it("Removes", () => {
    expect(remove("ipsum", initialArray, (a, b) => a === b)).toMatchSnapshot();
  });

  it("Removes at index", () => {
    expect(removeAtIndex(initialArray, 1)).toMatchSnapshot();
  });

  it("Matches", () => {
    expect(isSelected("lorem", initialArray, (a, b) => a === b)).toBe(true);
    expect(isSelected("sit", initialArray, (a, b) => a === b)).toBe(false);
  });

  it("Toggles", () => {
    expect(toggle("lorem", initialArray, (a, b) => a === b)).toMatchSnapshot();
    expect(toggle("sit", initialArray, (a, b) => a === b)).toMatchSnapshot();
  });

  it("Moves", () => {
    expect(move("lorem", initialArray, (a, b) => a === b, 1)).toMatchSnapshot();
  });
});
