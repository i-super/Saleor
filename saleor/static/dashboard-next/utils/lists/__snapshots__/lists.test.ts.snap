// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly calculates output arrays Adds 1`] = `
Array [
  "lorem",
  "ipsum",
  "dolor",
  "sit",
]
`;

exports[`Properly calculates output arrays Adds at index 1`] = `
Array [
  "lorem",
  "ipsum",
  "sit",
  "dolor",
]
`;

exports[`Properly calculates output arrays Moves 1`] = `
Array [
  "ipsum",
  "lorem",
  "dolor",
]
`;

exports[`Properly calculates output arrays Removes 1`] = `
Array [
  "lorem",
  "dolor",
]
`;

exports[`Properly calculates output arrays Removes at index 1`] = `
Array [
  "lorem",
  "dolor",
]
`;

exports[`Properly calculates output arrays Toggles 1`] = `
Array [
  "ipsum",
  "dolor",
]
`;

exports[`Properly calculates output arrays Toggles 2`] = `
Array [
  "lorem",
  "ipsum",
  "dolor",
  "sit",
]
`;

exports[`Properly calculates output arrays Updates at index 1`] = `
Array [
  "lorem",
  "amet",
  "dolor",
]
`;
